### PR TITLE
various net-related fixes

### PR DESCRIPTION
**Pull request by ArsenArsen**

### CODE REVIEW

File: options/posix/generic/arpa-inet-stubs.cpp [hunk 1]
```diff
@@ -65,9 +65,13 @@ in_addr_t inet_addr(const char *) {
 	__ensure(!"Not implemented");
 	__builtin_unreachable();
 }
-char *inet_ntoa(struct in_addr) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+char *inet_ntoa(struct in_addr addr) {
+	thread_local static char buffer[16];
```
Comment (avdgrinten) [inline options/posix/generic/arpa-inet-stubs.cpp:69]: Is this buffer big enough? 4*4+1 = 17.

File: options/posix/generic/arpa-inet-stubs.cpp [hunk 2]
```diff
@@ -65,9 +65,13 @@ in_addr_t inet_addr(const char *) {
...
+	int proper = htonl(addr.s_addr);
```
Comment (avdgrinten) [inline options/posix/generic/arpa-inet-stubs.cpp:70]: This should be a `uint32_t`.

File: options/posix/generic/arpa-inet-stubs.cpp (hunk 1 above)
Comment (ArsenArsen) [reply options/posix/generic/arpa-inet-stubs.cpp:69]: four groups of up to three numbers, three periods, and a null: `4 * 3 + 3 + 1 = 16`

Comment (avdgrinten) [reply options/posix/generic/arpa-inet-stubs.cpp:69]: Add a comment (it's not so hard to see but also not totally trivial).

**Comment by avdgrinten** [comment]
LGTM. Blocked on the corresponding Managarm changes.

**Pull request merged**